src/plugins/validation/2and3/semantic-validators/items-required-for-array-objects.js

```diff
@@ -39,11 +40,40 @@ module.exports.validate = function({ jsSpec }) {
       if (Array.isArray(obj.required)) {
         obj.required.forEach((requiredProp, i) => {
           if (!obj.properties || !obj.properties[requiredProp]) {
-            messages.addMessage(
-              path.concat([`required[${i}]`]).join('.'),
-              "Schema properties specified as 'required' must be defined",
-              'error'
-            );
+            // For allOf, ensure atleast one of the child objects has the required property
+            if (obj.allOf) {
```

Comment:
Might be a good idea to verify that this is an array. I know it is supposed to be according to the spec, but it would add some type safety since the code will crash on `forEach` otherwise.